app/controllers/grade_entry_forms_controller.rb

```diff
@@ -299,4 +299,11 @@ def switch
       redirect_to student_interface_course_grade_entry_form_path(current_course, params[:id])
     end
   end
+
+  def destroy
+    @grade_entry_form = record
+    @grade_entry_form.destroy
```

Comment:
You should check whether the deletion was successful or not, and flash an error message if the grade entry form could not be deleted.

You can test this case in the UI by making the Delete button (temporarily) appear even when it shouldn't, and then clicking on it.